packages/iris-grid/src/mousehandlers/IrisGridColumnSelectMouseHandler.ts

```diff
@@ -26,7 +26,8 @@ class IrisGridColumnSelectMouseHandler extends GridMouseHandler {
   }
 
   private isValidColumn(tableColumn: Column | null): boolean {
-    if (tableColumn == null) {
+    const { model } = this.irisGrid.props;
+    if (tableColumn == null || !model.isFilterable(tableColumn.index)) {
```

Comment:
So `column.index` JS API is actually deprecated and we should remove all uses of it: https://github.com/deephaven/web-client-ui/issues/965
Instead we should be storing/tracking by name, and then looking up the column by name (`model.getColumnIndexByName`, so this would be `model.isFilterable(model.getColumnIndexByName(tableColumn.name))`).
Side note, maybe we could allow `ModelIndex | ColumnName` to be passed in to `model.isFilterable` for convenience, and then if a `ColumnName` is passed in then convert that to model index automatically.